### ISSUE TITLE
Fix timing of existing Arellano code and experiment with JAX version

### ISSUE THREAD

**Issue by jstac** [QuantEcon/lecture-python-advanced.myst#107]
Thanks to @qbatista for pointing out the timing issue, as per [arellano_timing.pdf](https://github.com/QuantEcon/lecture-python-advanced.myst/files/8236984/arellano_timing.pdf)

This needs to be fixed.  

Thanks also for providing JAX code, which uses the correct timing.

**Comment by jstac** [comment]
The code is now fixed with numba but we should probably add a full JAX version, which I've implemented here.

https://github.com/QuantEcon/cbc_workshops/tree/main/arellano

(I wrote the JAX version from scratch as an exercise but it should be similar to the one that Quentin sent to @thomassargent30 .  The output agrees with the numba version and it's one or two orders of magnitude faster.)

Perhaps @Smit-create and I can finish this up when he visits ANU.  A open question is whether or not to set up an independent set of lectures for JAX code, compiled on the GPU (CC @mmcky ).
